Keep bookmark returned after auto-commit tx

lutovich opened the pull request:
Auto-commit transactions return bookmarks in Bolt V3. They were previously unused. This PR makes driver extract such bookmarks and expose them via `Session#lastBookmark()`. It then possible for explicit and auto-commit transactions within the same session to be chained with bookmarks.

lutovich committed:
Keep bookmark returned after auto-commit tx

Auto-commit transactions return bookmarks in Bolt V3. They were
previously unused. This commit makes driver extract such bookmarks
and expose them via `Session#lastBookmark()`. It then possible for
explicit and auto-commit transactions within the same session to be
chained with bookmarks.

zhenlineo commented:
The concept of BookmarkHolder is really good. Wondering if it also makes sense to make Tx to get and set bookmark via BookmarkHolder too.

lutovich committed:
Do not keep bookmark in transaction

Pass it directly to the session, when the transaction is committed.
This simplifies `ExplicitTransaction` a bit by removing all bookmark
handling from it.

lutovich commented:
@zhenlineo I removed all the bookmark handling from transaction :)
`ExplicitTransaction` will now propagate bookmark directly to the session after commit.
